yt_dlp/extractor/instagram.py

```diff
@@ -329,9 +329,10 @@ def get_count(keys, kind):
         }
 
 
-class InstagramPlaylistIE(InfoExtractor):
+class InstagramPlaylistIE(InstagramIE):
```

Comment:
It is recommended to make an `InstagramBaseIE` instead to avoid accidentally sharing undesirable properties. I can see you have made sure to overload everything manually, but that is the wrong approach

Also, in the login code, make sure the token is saved as a class var of the base class (not an instance var) so that it is properly shared b/w extractors. You can test this by checking that a playlist download only causes a single login

Reply:
Created `InstagramBaseIE` but it still causes two logins.

Reply:
that's why I said this

> Also, in the login code, make sure the token is saved as a class var of the base class (not an instance var) so that it is properly shared b/w extractors.

see my other suggestion